### ISSUE TITLE
Add support for component pedigree

### ISSUE THREAD

**Issue by stevespringett** [CycloneDX/specification#6]
This is an enhancement proposal to the CycloneDX specification to support `ancestors`. This would be useful for forks of projects that have modifications in which the identification of the component has changed. 

The `modified` flag can be used if the component metadata has not changed. For example, if Apache Tomcat was modified but the group, name, version, etc remained the same. If this data has been modified, it may be impossible to perform accurate analysis with the modified data.

The proposal is to add ancestors in which a component descends from. Ancestors is a list of 0-n components, so if a component used the Maven shade plugin for example, this would aggregate one or more components into a single one. In the following example, Apache Tomcat has been modified by Acme Inc.

With ancestors, it would be possible to track the origin of each component so that accurate analysis can be performed at each level of the tree.

For example:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bom xmlns="http://cyclonedx.org/schema/bom/1.1" version="1">
    <components>
        <component type="application">
            <publisher>Acme Inc</publisher>
            <group>com.acme</group>
            <name>tomcat-catalina</name>
            <version>9.0.14</version>
            <licenses>
                <license>
                    <id>Apache-2.0</id>
                </license>
            </licenses>
            <purl>pkg:maven/com.acme/tomcat-catalina@9.0.14?packaging=jar</purl>
            <modified>false</modified>
            <ancestors>
                <component type="application">
                    <publisher>Apache</publisher>
                    <group>org.apache.tomcat</group>
                    <name>tomcat-catalina</name>
                    <version>9.0.14</version>
                    <licenses>
                        <license>
                            <id>Apache-2.0</id>
                        </license>
                    </licenses>
                    <purl>pkg:maven/org.apache.tomcat/tomcat-catalina@9.0.14?packaging=jar</purl>
                    <modified>false</modified>
                </component>
            </ancestors>
        </component>
    </components>
</bom>
```

**Comment by stevespringett** [comment]
Although the second option is a bit more verbose (pedigree > ancestors > components) I think it offers the most flexibility and future-proof design.

For example, it would be possible to document every commit that lead to the creation of a given component with something like:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bom xmlns="http://cyclonedx.org/schema/bom/1.1" version="1">
    <components>
        <component type="application">
            <publisher>Acme Inc</publisher>
            <group>com.acme</group>
            <name>tomcat-catalina</name>
            <version>9.0.14</version>
            <licenses>
                <license>
                    <id>Apache-2.0</id>
                </license>
            </licenses>
            <purl>pkg:maven/com.acme/tomcat-catalina@9.0.14?packaging=jar</purl>
            <modified>false</modified>
            <pedigree>
                <ancestors>
                    <component type="application">
                        <publisher>Apache</publisher>
                        <group>org.apache.tomcat</group>
                        <name>tomcat-catalina</name>
                        <version>9.0.14</version>
                        <licenses>
                            <license>
                                <id>Apache-2.0</id>
                            </license>
                        </licenses>
                        <purl>pkg:maven/org.apache.tomcat/tomcat-catalina@9.0.14?packaging=jar</purl>
                        <modified>false</modified>
                    </component>
                </ancestors>
                <commits>
                    <commit>
                        <hash alg="sha-1">7638417db6d59f3c431d3e1f261cc637155684cd</hash>
                        <url>https://location/to/7638417db6d59f3c431d3e1f261cc637155684cd</url>
                        <author>
                            <timestamp>2014-11-07T22:01:45Z</timestamp>
                            <name>John Doe</name>
                            <email>jdoe@example.com</email>
                        </author>
                        <committer>
                            <timestamp>2014-11-07T22:01:45Z</timestamp>
                            <name>John Doe</name>
                            <email>jdoe@example.com</email>
                        </author>
                        <message>Initial commit</message>
                    </commit>
                </commits>
            </pedigree>
        </component>
    </components>
</bom>
```

The above BOM describes a forked version of a Tomcat component. It identifies the component in which it originates from as well as the commits that make it different.

**Comment by stevespringett** [comment]
Included in CycloneDX 1.1
